File: Java/Shirt.java
/*
Bismillah Saya Themy Sabri Syuhada dengan NIM 2203903.
Demi keberkahan-Nya, saya berjanji mengerjakan
latihan praktikum 2 DPBO dengan jujur dan
tidak melakukan kecurangan seperti yang telah dispesifikasikan.
Aamiin.
*/

// Shirt.java
public class Shirt extends Clothing {
    protected String color;
    protected String sleeveType;

    public Shirt(int idProduct, String name, String brand, double price, String size, String material, String gender,
            String color, String sleeveType) {
        super(idProduct, name, brand, price, size, material, gender);
        this.color = color;
        this.sleeveType = sleeveType;
    }

    public String getColor() {
        return this.color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getSleeveType() {
        return this.sleeveType;
    }

    public void setSleeveType(String sleeveType) {
        this.sleeveType = sleeveType;
    }

    public String toString() {
        return super.toString() + String.format(" %-8s| %-10s |", color, sleeveType);
    }
}

File: PHP/Shirt.php
<?php

include 'Clothing.php'; 

class Shirt extends Clothing
{
    protected $color;
    protected $sleeveType;

    public function __construct($idProduct, $name, $brand, $price, $size, $material, $gender, $color, $sleeveType)
    {
        parent::__construct($idProduct, $name, $brand, $price, $size, $material, $gender);
        $this->color = $color;
        $this->sleeveType = $sleeveType;
    }

    public function getColor()
    {
        return $this->color;
    }

    public function setColor($color)
    {
        $this->color = $color;
    }

    public function getSleeveType()
    {
        return $this->sleeveType;
    }

    public function setSleeveType($sleeveType)
    {
        $this->sleeveType = $sleeveType;
    }

    public function __toString()
    {
        return parent::__toString() . sprintf(" %-8s| %-10s |", $this->color, $this->sleeveType);
    }
}
